src/rules/javascript/no_document_write.js

```diff
@@ -0,0 +1,29 @@
+import { NO_DOCUMENT_WRITE } from 'messages/javascript';
+import { getNodeReferenceName } from 'utils';
+
+export default function(context) {
+  return {
+    CallExpression: function(node) {
+      var objectName;
+      if (typeof node.callee.object !== 'undefined' &&
+          node.callee.object.type === 'Identifier') {
+
+        objectName = getNodeReferenceName(context, node.callee.object);
+        if (typeof objectName === 'undefined') {
+          return;
+        }
+      }
+
+
+      // check to see if the node is document.write()
+      if (typeof node.callee.property !== 'undefined' &&
```

Comment:
I think you can probably DRY up these ifs a bit. The first two checks are the same as the if starting line 7.